sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/_tests_/e2e/confirmation-of-enrollment.institutions.denyConfirmationOfEnrollment.e2e-spec.ts

```diff
@@ -83,6 +84,34 @@ describe("ConfirmationOfEnrollmentInstitutionsController(e2e)-denyConfirmationOf
     expect(declinedCOE.coeDeniedReason.id).toBe(coeDenyReasonId);
   });
 
+  it("Should not decline the COE when user is read-only.", async () => {
+    // Arrange
+    const { institution: collegeE } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeEReadOnlyUser,
+    );
+    const collegeELocation = createFakeInstitutionLocation({
+      institution: collegeE,
+    });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeEReadOnlyUser,
+      collegeELocation,
+      InstitutionUserTypes.readOnlyUser,
+    );
```

Comment:
This code is reused many times along many E2E tests and it would be good to have it moved to a factory, maybe inside the `institution-auth-helpers.ts`.